source/client.c

```diff
@@ -2484,7 +2501,7 @@ static enum aws_mqtt_client_request_state s_publish_send(uint16_t packet_id, boo
         }
 
         if (aws_channel_slot_send_message(task_arg->connection->slot, message, AWS_CHANNEL_DIR_WRITE)) {
-
+            /* TODO: NOT really agree on retry for this failure */
             aws_mem_release(message->allocator, message);
             return AWS_MQTT_CLIENT_REQUEST_ERROR;
         }
```

Comment:
this can fail if the channel is shutting down
it should probably do like subscribe and just pretend things are ok